pkg/virt-launcher/virtwrap/network/ndp/connection.go

```diff
@@ -0,0 +1,126 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2020 Red Hat, Inc.
+ *
+ */
+
+package ndp
+
+import (
+	"fmt"
+	"net"
+	"os"
+
+	"github.com/mdlayher/ndp"
+	"golang.org/x/net/ipv6"
+)
+
+const (
+	chkOff  = 2
+	maxHops = 255
+)
+
+// A NDPConnection instruments a system.Conn and adds retry functionality for
+// receiving / sending NDP messages on a given interface.
+type NDPConnection struct {
+	iface      *net.Interface
+	rawConn    *net.IPConn
+	conn       *ipv6.PacketConn
+	controlMsg *ipv6.ControlMessage
+}
+
+// Return an NDPConnection bound to the chosen interface.
+func NewNDPConnection(ifaceName string) (*NDPConnection, error) {
+	iface, err := net.InterfaceByName(ifaceName)
+	if err != nil {
+		return nil, fmt.Errorf("could not find interface %s: %v", ifaceName, err)
+	}
+
+	listenAddr := &net.IPAddr{
+		IP:   net.IPv6unspecified,
+		Zone: ifaceName,
+	}
+	icmpListener, err := net.ListenIP("ip6:ipv6-icmp", listenAddr)
+	if err != nil {
+		return nil, fmt.Errorf("could not listen to ip6:ipv6-icmp on addr %s: %v", listenAddr.String(), err)
+	}
+
+	ipv6Conn := ipv6.NewPacketConn(icmpListener)
+
+	_ = ipv6Conn.SetHopLimit(maxHops)          // as per RFC 4861, section 4.1
```

Comment:
Are there any outgoing non-milticast messages? Why there is a need to et both the HopLimit and the MulticastHopLimit?

Reply:
Why do you ignore the possible returned errror?